Sources/SourceKitD/DynamicallyLoadedSourceKitD.swift

```diff
@@ -104,6 +119,8 @@ public actor DynamicallyLoadedSourceKitD: SourceKitD {
   }
 
   public nonisolated func log(request: SKDRequestDictionary) {
+    // `log(request:)` is called in `SourceKitD.send` before the request is sent, so we're hooking in here.
+    testHooks.sourcekitdRequestDidStart?(request)
```

Comment:
Any reason not to put this in `func send`? I guess to avoid exposing it in the protocol?

Reply:
I was thinking because protocols can’t have stored properties. But somehow didn’t occur to me before that we could just make the test hooks a protocol requirement… 🤦🏽